### PR TITLE
feat: add support for session storage

### PR DESCRIPTION
**Pull request by MoumitaM**
## PR Description

1. Session storage is added as a storage type for storing user data and existing storage.
2. Mode changes in multiple modules to add it as a configurable storage type for cookie-less tracking feature.

## Linear task (optional)

Linear task link

## Cross Browser Tests

Please confirm you have tested for the following browsers:

- [ ] Chrome
- [ ] Firefox
- [ ] IE11

## Security

- [ ] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

### CODE REVIEW

**Review by saikumarrs — APPROVED** [review]

File: packages/analytics-js/src/services/StoreManager/storages/defaultOptions.ts [hunk 1]
```diff
@@ -22,8 +22,17 @@ const getDefaultLocalStorageOptions = (): ILocalStorageOptions => ({
   enabled: true,
 });
 
+const getDefaultSessionStorageOptions = (): ILocalStorageOptions => ({
```
Comment (bardisg) [inline packages/analytics-js/src/services/StoreManager/storages/defaultOptions.ts:25]: should type be ISessionStorageOptions instead?

File: packages/analytics-js/src/services/StoreManager/storages/storageEngine.ts [hunk 2]
```diff
@@ -57,23 +59,33 @@ const configureInMemoryStorageEngine = (options: Partial<IInMemoryStorageOptions
   defaultInMemoryStorage.configure(options);
 };
 
+/**
+ * Configure in memory storage singleton
```
Comment (bardisg) [inline packages/analytics-js/src/services/StoreManager/storages/storageEngine.ts:63]: comment is misleading should be session storage instead

File: packages/analytics-js/src/services/StoreManager/storages/defaultOptions.ts (hunk 1 above)
Comment (MoumitaM) [reply packages/analytics-js/src/services/StoreManager/storages/defaultOptions.ts]: Addressed

File: packages/analytics-js/src/services/StoreManager/storages/storageEngine.ts (hunk 2 above)
Comment (MoumitaM) [reply packages/analytics-js/src/services/StoreManager/storages/storageEngine.ts]: Addressed

**Review by saikumarrs — APPROVED** [review]

**Review by bardisg — APPROVED** [review]

**Review by saikumarrs — APPROVED** [review]

**Pull request merged**